crawlfrontier/contrib/backends/sqlalchemy/__init__.py

```diff
@@ -122,7 +122,8 @@ def frontier_stop(self):
 
     def add_seeds(self, seeds):
         for seed in seeds:
-            db_page, _ = self._get_or_create_db_page(url=seed.url, fingerprint=seed.meta['fingerprint'],
+            url, fingerprint, _ = self.manager.canonicalsolver.get_canonical_url(seed)
```

Comment:
I see this pair of lines together all the time - `get_canonical_url` followed by `_get_or_create_db_page`. should there be a single method to do both?